scipy/stats/stats.py

```diff
@@ -1507,12 +1604,53 @@ def itemfreq(a):
 
     """
     items, inv = np.unique(a, return_inverse=True)
-    freq = np.bincount(inv)
+    freq = np.bincount(inv, weights=weights)
     return np.array([items, freq]).T
 
 
+def collapse_weights(a, weights=None, axis=None):
+    """
+    Collapse a weigthed representation of an array.
+
+    Parameters
+    ----------
+    a : array_like
+        A ndarray of values.
+    axis : int, optional
+        Axis along which uniques are computed, and weights are valid. Default
+        is None. If  None, flatten `a` first.
+    weights : array_like, optional
+        The weights for each record along `axis` in `a`. Default is None,
+        which gives each record a weight of 1.0.
+
+    Returns
+    -------
+    uniq_a : ndarray
+        An ndarray of the sorted record in `a`, without duplicates.
+    uniq_w : array
+        The weights for each record in `uniq_a` along `axis`d that
+        make it representationally equivalent.
+    """
+    a, weights, axis = _chk_weights([a], weights=weights, axis=axis, pos_only=True,
+                                    force_weights=True, simplify_weights=False)
+    sort_axes = [0] * a.ndim
+    sort_axes[axis] = slice(None)
+    sort_ix = np.argsort(a, axis=axis)[sort_axes]
+    sorted_a = np.take(a, sort_ix, axis=axis)
+    new_a = np.diff(sorted_a, axis=axis)
+    if a.ndim > 1:
+        not_axes = tuple(i for i in range(a.ndim) if i != axis)
+        new_a = new_a.any(axis=not_axes)
+    new_a = np.append(0, np.where(new_a)[0]+1)
+
+    uniq_a = np.take(sorted_a, new_a, axis=axis)
+    cum_w = np.append(0, weights[sort_ix].cumsum())
+    uniq_w = np.diff(cum_w[np.append(new_a, -1)])
+    return uniq_a, uniq_w
```

Comment:
If axis is not None, and ndim > 1, then it considers a "record" to consist of all of the axes which are not the axis specified. It looks for uniques among "records", and sums up their weight. It assumes that `a.shape[axis] == weights.size`